PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java

```diff
@@ -309,4 +314,67 @@ public void removeTabs() {
             appBar.removeViewAt(1);
         }
     }
+
+    private void getPermission(int code) {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) !=
+                PackageManager.PERMISSION_GRANTED ||
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) !=
+                        PackageManager.PERMISSION_GRANTED) {
+
+            ActivityCompat
+                    .requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION,
+                            Manifest.permission.ACCESS_COARSE_LOCATION}, code);
+        } else {
+            Fragment fragment = null;
+            switch(code) {
+                case CODE_MAP:
+                    fragment = new MapFragment();
+                    break;
+                case CODE_TRANSIT:
+                    fragment = new TransitFragment();
+                    break;
+            }
+            if (fragment != null) {
+                FragmentManager fragmentManager = getSupportFragmentManager();
+                fragmentManager.beginTransaction()
+                        .replace(R.id.content_frame, fragment)
+                        .addToBackStack(null)
+                        .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                        .commit();
+            }
+            mDrawerLayout.closeDrawer(mDrawerList);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        if (grantResults.length == 0 || grantResults[0] == PackageManager.PERMISSION_DENIED) {
+            //um so couldn't get permission...
+            return;
```

Comment:
Would be good to add a notification snackbar or toast explaining that permissions are required to use the map/transit functions of the app instead of silently closing